### PR TITLE
Upgrade iota.go for recent Curl improvements

### PR DESCRIPTION
**Pull request by Wollac**
# Description

This PR upgrades iota.go to include the recent Curl improvements:
- Use `iota.go/curl` and `iota.go/curl/bct` instead of the much less performant hive.go versions
- Migrate `batchhasher` from hive.go into hornet.
- Optimize `batchhasher` and add testcases.

## Type of change

- [x] Enhancement (non-breaking change which adds functionality)

# How Has This Been Tested?

- Added unit tests.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have tested my code extensively
- [x] I have selected the `develop` branch as the target branch

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -55,3 +55,5 @@ require (
 	google.golang.org/grpc v1.32.0 // indirect
 	gopkg.in/ini.v1 v1.61.0 // indirect
 )
+
+replace github.com/iotaledger/iota.go => github.com/Wollac/iota.go v1.0.0-beta.9.0.20200923185057-e07ab2b9ea38
```
Comment (Wollac) [inline go.mod]: The usage of the fork needs to be removed as soon as https://github.com/iotaledger/iota.go/pull/195 is merged

File: pkg/batcher/batcher.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+	return c.workerCount
+}
+
+// BatchSize returns the maximum batch size.
+func (c *Curl) BatchSize() int {
+	return bct.MaxBatchSize
+}
+
+// Close stops the batched Curl hasher.
+// It blocks until all remaining jobs have been processed or canceled. Successive cals of Close are ignored.
```
Comment (muXxer) [inline pkg/batcher/batcher.go]: Small typo "calls"

File: pkg/model/hornet/hash.go [hunk 3]
```diff
@@ -3,46 +3,103 @@ package hornet
...
 
+	"github.com/iotaledger/iota.go/consts"
+	"github.com/iotaledger/iota.go/encoding/t5b1"
 	"github.com/iotaledger/iota.go/trinary"
 )
 
 var (
 	// NullHashBytes is the binary hash of the genesis transaction.
-	NullHashBytes = make(Hash, 49)
+	NullHashBytes = make(Hash, hashBytesSize)
```
Comment (muXxer) [inline pkg/model/hornet/hash.go]: Does go accept that? hashBytesSize is declared in the next line...

**Review by muXxer — APPROVED** [review]

File: pkg/model/hornet/hash.go (hunk 3 above)
Comment (Wollac) [reply pkg/model/hornet/hash.go]: Yes, the compiler performs a dependency evaluation for initialization:
> Within a package, package-level variable initialization proceeds stepwise, with each step selecting the variable earliest in declaration order which has no dependencies on uninitialized variables. 
https://golang.org/ref/spec#Package_initialization

Should I rather change the order, though?

Comment (muXxer) [reply pkg/model/hornet/hash.go]: Oh didn't knew that.
Pls change :)

**Review by muXxer — APPROVED** [review]

**Pull request merged**